### PR TITLE
Add unit tests, improve a dozen other things

### PR DESCRIPTION
**Pull request by piontec**
Another milestone PR to set main things in place. Next ones should finally be smaller.

This one includes:

- full test coverage of core basic classes
- support for `--skip-steps` option
- dabs.sh script should not break file permissions anymore
- bug fixes

### CODE REVIEW

**Review by ubergesundheit — DISMISSED** [review]
All in all LGTM!

Some things in Dockerfile could be improved but I could've mentioned this in the last review..

If you want, I can issue a PR for things I think could be improved in the Dockerfile

**Comment by piontec** [comment]
> All in all LGTM!
> 
> Some things in Dockerfile could be improved but I could've mentioned this in the last review..
> 
> If you want, I can issue a PR for things I think could be improved in the Dockerfile

Sure I do :) Any contributions are welcome.

File: run-tests-in-docker.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -e
+
+#if [ ${USE_UID:-0} -ne 0 ] && [ ${USE_GID:-0} -ne 0 ]; then
+#  groupadd -f -g $USE_UID abs
+#  useradd -g $USE_GID -M -l -u $USE_UID abs -d $ABS_DIR -s /bin/bash || true
+#fi
+
+#chown -R $USE_UID:$USE_GID $ABS_DIR
+#sudo --preserve-env=PYTHONPATH,PATH -g "#$USE_GID" -u "#$USE_UID" -- python -m app_build_suite $@
+#pipenv run pre-commit run --all-files
```
Comment (ubergesundheit) [inline run-tests-in-docker.sh:10]: Is this file required at all? You could at least remove the commented out lines

Comment (piontec) [reply run-tests-in-docker.sh]: Yes, it's WIP - will clean it up :)
